Minor Pre launch fixes

amitojsingh opened the pull request:
- enable sync scroll by default where available.
- Highlighted the panktee and scroll to start of the panktee
- fix the bottom navigation to show both reader and audio icons when moving from the Reader to Settings, with 'Settings' highlighted by default.

amitojsingh committed:
Enhance testing and functionality of BottomNavigation component

- Updated BottomNavigation to utilize the useNavigation hook for improved navigation handling.
- Simplified props by removing the navigation prop and managing navigation state internally.
- Added logic to conditionally display navigation items based on the previous route, enhancing user experience when transitioning between screens.
- Improved test coverage for BottomNavigation, including scenarios for navigating from different routes and ensuring correct audio state management.
- Mocked useNavigation in tests to ensure accurate navigation behavior during component rendering.

amitojsingh committed:
Implement early return in handleTrackSelect for null or undefined selectedTrack in AudioPlayer component

Gauravjeetsingh commented on src/common/components/BottomNavigation/index.jsx:
```diff
@@ -1,20 +1,28 @@
+  const getCurrentRouteName = () => {
+    const navState = navigation.getState();
+    return navState?.routes[navState?.index]?.name;
+  };
```
The `getCurrentRouteName` function is recreated on every render. Consider memoizing it or using `useCallback` if it's called frequently.

amitojsingh replied:
fixed

Gauravjeetsingh reviewed (approved):
Minor thing, can be merged after that. Approving this.

amitojsingh committed:
Refactor BottomNavigation component to use useCallback for getCurrentRouteName function

- Updated getCurrentRouteName to utilize useCallback for improved performance and to prevent unnecessary re-renders.
- Enhanced the import statement to include useCallback from React, aligning with best practices for functional components.
